frontend/src/SigninButton/SigninButton.tsx

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import Button from "@material-ui/core/Button";
+import "./SigninButton.css";
+
+interface ISigninButtonProps {
+  isSignedIn: boolean;
+  onClick: () => void;
+}
+
+function SigninButton(props: ISigninButtonProps) {
+  return (
+    <div className="SigninButton">
+      {props.isSignedIn ? (
```

Comment:
What about making the ternary expression `a ? b : c` finer-grained? For example, "SIGN IN" and "SIGN OUT" could be extracted to strings in a map then we would use `...>props.isSignedIn ? buttonText.signOut : buttonText.signIn<...`